ocrd_eval.md

```diff
@@ -0,0 +1,372 @@
+
+Given a region A with an area $area_1$, a region B with the area $area_2$, and their overlap (or intersection) $area_o$, the IoU can then be expressed as
+
+$IoU = \frac{area_o}{area_1+area_2-area_o}$
+
+where $area_1+area_2-area_o$ expresses the union of the two regions ($area_1+area_2$) while not counting the overlapping area twice.
+
+The IoU ranges between 0 (no overlap at all) and 1 (the two regions overlap perfectly). Users executing object detection can choose a [threshold](#Threshold) that defines which degree of overlap must be given to define a prediction as correct. If e.g. a threshold of 0.6 is chosen, all prediction that have an IoU of 0.6 or higher are correct.
+
+In OCR-D we use IoU to measure how well segments on a page are recognized during the segmentation step. The area of one region represents the area identified in the Ground Truth, while the second region represents the area identified by an OCR-D processor.
+
+### Resource Utilization
+
+Last but not least, it is important to collect information about the resource utilization of each processing step, so that informed decisions can be made when e.g. having to decide between results quality and throughput speed.
+
+#### CPU Time
+
+CPU time is the time taken by the CPU to process an instruction. It does not include idle time.
+
+#### Wall Time
+
+Wall time (or elapsed time) is the time taken by a processor to process an instruction including idle time.
+
+#### I/O
+
+I/O (input / output) is the number of bytes read and written during a process.
+
+#### Memory Usage
+
+Memory usage is the number of bytes the process allocates in memory (RAM).
+
+#### Disk Usage
+
+Disk usage is the number of bytes the process allocates on hard disk.
+
+### Unicode Normalization
+
+In Unicode there can be multiple ways to express characters that have multiple components, such as a base letter and an accent. For evaluation it is essential that both Ground Truth and OCR results are normalized *in the same way* before evaluation.
+
+For example, the letter `ä` can be expressed directly as `ä` (`U+00E4` in Unicode) or as a combination of `a` and `◌̈` (`U+0061 + U+0308`). Both encodings are semantically equivalent but technically different.
+
+Unicode has the notion of *normalization forms* to provide canonically normalized text. The most common forms are *NFC* (Normalization Form Canonical Composed) and *NFD* (Normalization Form Canonical Decomposed). When a Unicode string is in NFC, all decomposed codepoints are replaced with their decomposed equivalent (e.g. `U+0061 + U+0308` to `U+00E4`). In an NFD encoding, all decomposed codepoints are replaced with their composed equivalents (e.g. `U+00E4` to `U+0061 + U+0308`).
+
+<!-- There's also NFKC and NFKD - necessary to explain? -->
+
+In accordance with the concept of [GT levels in OCR-D](https://ocr-d.de/en/gt-guidelines/trans/trLevels.html), it is preferable for strings to be normalized as NFC.
```

Comment:
I'm not sure what you aim at here, could you elaborate please? (Maybe I have a terminology problem. Reading "equivalency" I think of equivalence classes etc., but they don't have levels.)

Reply:
Either by normalising both sides to OCR-D GT level 2 (or 1). Or by passing equivalence classes (zero edit cost rules) to the distance metric.

For example, naively, an umlaut error (`u` instead of `ü` or `uͤ`), or a punctuation error (`"` instead of `“`), will have the same cost as any other error. But they might not be as relevant as others.

Reply:
Since @kba wrote the better part of this paragraph – what do you think?

Reply:
We could also track this in a new issue, or move to the `Metrics Not In Use Yet` section.